Dockerfile

```diff
@@ -1,6 +1,16 @@
 FROM eclipse-temurin:19-jre-jammy
```

Comment:
maybe should be updated to 21?

Reply:
Yes, but maybe that update can be done in its own pullrequest where all dependencies for java21 is updated and then I can rebase this pullrequest on main to get these changes? Might be safer since this PR also includes new tests and some other updates to build files like Dockerfile and release.yml?

Reply:
Yes, that should be the way to do it, good input on that!